src/index.js

```diff
@@ -1,100 +1,93 @@
-import { join, dirname, extname, basename } from 'path'
-import { parse } from 'babylon';
+import { join, dirname, extname, basename } from 'path';
+import loaderUtils from 'loader-utils';
 
 import traverse from './traverse';
+import VirtualModulePlugin from './VirtualModulePlugin';
 
-function parseSource(src) {
-  return parse(src, {
-    sourceType: 'module',
-    plugins: [
-      'asyncFunctions',
-      'jsx',
-      'flow',
-      'classConstructorCall',
-      'doExpressions',
-      'trailingFunctionCommas',
-      'objectRestSpread',
-      'decorators',
-      'classProperties',
-      'exportExtensions',
-      'exponentiationOperator',
-      'asyncGenerators',
-      'functionBind',
-      'functionSent'
-    ]
-  })
-}
-
-function collectStyles(src) {
-  let styles = [];
+function collectStyles(src, tagName = 'css') {
+  const styles = [];
 
   // quick regex as an optimization to avoid parsing each file
-  if (!src.match(/css`([\s\S]*?)`/gmi)) {
+  if (!src.match(new RegExp(`${tagName}\`([\\s\\S]*?)\``, 'gmi'))) {
```

Comment:
i was leaving it in "Why would you do this" sort of thing, least until someone complains